virtio/transport/mmio.c

```diff
@@ -0,0 +1,158 @@
+/*
+ * Copyright 2025, UNSW
+ * SPDX-License-Identifier: BSD-2-Clause
+ */
+
+#include <sddf/virtio/transport/common.h>
+#include <sddf/virtio/transport/mmio.h>
+#include <sddf/resources/device.h>
+
+static bool check_magic(virtio_mmio_regs_t *regs)
+{
+    return regs->MagicValue == 0x74726976;
+}
+
+static bool check_device_id(virtio_mmio_regs_t *regs, virtio_mmio_device_id_t id)
+{
+    return regs->DeviceID == id;
+}
+
+static uint32_t get_version(virtio_mmio_regs_t *regs)
+{
+    return regs->Version;
+}
+
+static volatile virtio_mmio_regs_t *get_regs(device_resources_t *device_resources)
+{
+    return (volatile virtio_mmio_regs_t *)device_resources->regions[0].region.vaddr;
+}
+
+bool virtio_transport_probe(device_resources_t *device_resources, virtio_device_handle_t *device_handle_ret)
```

Comment:
This should take in the the virtIO device ID by the caller, then we don't have to make a new case for each new device added.